### PR TITLE
[ONNX CI] nGraph master as docker image

### PR DESCRIPTION
**Pull request by mchrusci**
Updated CI workflow:
    * Periodically rebuilding nGraph master inside docker container and pushing it to registry
    * regular CI pull this docker image and build branch using prebuild master inside container as base

### CODE REVIEW

File: .ci/onnx/jenkins/base_image_builder/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// Set LABEL variable if empty or not declared
+try{ if(LABEL.trim() == "") {throw new Exception();} }catch(Exception e){LABEL="onnx && ci"}; echo "${LABEL}"
+try{ if(DOCKER_REGISTRY.trim() == "") {throw new Exception();} }catch(Exception e){DOCKER_REGISTRY="amr-registry.caas.intel.com"}; echo "${DOCKER_REGISTRY}"
```
Comment (postrational) [inline .ci/onnx/jenkins/base_image_builder/Jenkinsfile]: I don't think we should be exposing internal domain names in a public configuration file.

File: .ci/onnx/jenkins/utils/docker.sh [hunk 2]
```diff
@@ -15,9 +15,10 @@
 # the Materials, either expressly, by implication, inducement, estoppel or
 # otherwise. Any license under such intellectual property rights must be express
 # and approved by Intel in writing.
-readonly PARAMETERS=( 'name' 'version' 'container_name' 'volumes' 'env' 'ports' 'dockerfile_path' 'directory' 'options' 'tag' 'engine' 'frontend' 'new_tag' 'image_name' 'repository_type' 'build_cores_number')
+readonly PARAMETERS=( 'name' 'version' 'container_name' 'volumes' 'env' 'ports' 'dockerfile_path' 'directory' 'docker_registry'
+                      'options' 'tag' 'engine' 'frontend' 'new_tag' 'image_name' 'repository_type' 'build_cores_number')
 readonly WORKDIR="$(git rev-parse --show-toplevel)"
-readonly HUB_ADDRESS="hub.docker.intel.com"
+DOCKER_REGISTRY="amr-registry.caas.intel.com"
```
Comment (postrational) [inline .ci/onnx/jenkins/utils/docker.sh]: As above...

File: .ci/onnx/jenkins/utils/docker.sh [hunk 3]
```diff
@@ -40,21 +41,22 @@ docker.get_image_name() {
...
+        engine="/${engine}"
     fi
 
     if [ ! -z ${frontend} ]; then
-        frontend="_${frontend}"
+        frontend="/${frontend}"
     fi
 
-    echo "${HUB_ADDRESS}/aibt_${name,,}${repository_type,,}/${version,,}${engine,,}${frontend,,}:${tag}"
+    # amr-registry.caas.intel.com/aibt/aibt/ngraph_cpp/ubuntu_16_04/base:ci
```
Comment (postrational) [inline .ci/onnx/jenkins/utils/docker.sh]: As above...

File: .ci/onnx/jenkins/Jenkinsfile [hunk 4]
```diff
@@ -17,10 +17,11 @@
 
 // Set LABEL variable if empty or not declared
 try{ if(LABEL.trim() == "") {throw new Exception();} }catch(Exception e){LABEL="onnx && ci"}; echo "${LABEL}"
+try{ if(DOCKER_REGISTRY.trim() == "") {throw new Exception();} }catch(Exception e){DOCKER_REGISTRY="amr-registry.caas.intel.com"}; echo "${DOCKER_REGISTRY}"
```
Comment (postrational) [inline .ci/onnx/jenkins/Jenkinsfile]: One more place where we expose an internal URL.

Comment (mchrusci) [reply .ci/onnx/jenkins/Jenkinsfile]: Addressed

File: .ci/onnx/jenkins/base_image_builder/Jenkinsfile (hunk 1 above)
Comment (mchrusci) [reply .ci/onnx/jenkins/base_image_builder/Jenkinsfile]: Addressed

File: .ci/onnx/jenkins/utils/docker.sh (hunk 2 above)
Comment (mchrusci) [reply .ci/onnx/jenkins/utils/docker.sh]: Addressed

File: .ci/onnx/jenkins/utils/docker.sh (hunk 3 above)
Comment (mchrusci) [reply .ci/onnx/jenkins/utils/docker.sh]: Addressed

**Pull request merged**